### ISSUE TITLE
[Feature] Reddit Saved Posts with Gold

### ISSUE THREAD

**Issue by JustCampMan** [shadowmoose/RedditDownloader#29]
As you may know, reddit allows you to only see 1000 saved posts. But if you buy gold, this goes to 1000 per subreddit. I used your program to download the 1000 posts then learned of the limit and bought gold to be able to download more images. I unsaved the first 1000 posts.

Now the problem is, normally saved posts are accessible from "www.reddit.com/user/USERNAME/saved", but when you have gold, this section is still empty (in my case at least, because I unsaved the first 1000 posts). Now, I've to go to "www.reddit.com/user/USERNAME/saved?sr=SUBREDDITNAME" and this again allows me to see 1000 saved posts PER subreddit. But I can't seem to access this using your program.

I don't know if reddit API allows you to access saved posts like this, so I was thinking if you could also add a new source option, one that allows you to add LINKS from which it would download the posts. So I could give links, lets say "www.reddit.com/user/vargas/saved?sr=aww" and get posts from each subreddit one by one.

I would really appreciate it, thanks!

**Comment by JustCampMan** [comment]
I googled and apparently this works in praw. I don't know anythinig about python tho so can you make some sense of this?

s=r.user.get_saved(params={'sr':'aww'})

**Comment by shadowmoose** [comment]
Interesting. I didn't know that Gold extended access like that. I'll look into making the changes.

**Comment by JustCampMan** [comment]
I bought gold just for this, and I've like 28 days left. I'll love you
forever if you'd merge it by this week :c

**Comment by shadowmoose** [comment]
The WebUI development is underway, and brings with it a change to how Sources store settings. Because of this, it's easiest to augment your functionality into that branch. I'm stuck waiting on PRAW to add some unrelated auth functionality before I can roll that all out into a full release, but if you're willing to run a non-release build, you can use the ```webui``` branch.

As of commit 995fc6876f111feab39040bf9aba0b47a87f4b06, this functionality should be added to the ```user-upvoted-saved``` Source. At least, it all tests properly. I don't have an account with that amount of posts saved, so in-depth testing will have to wait a bit. You can download a zip archive frozen with that addition [here](https://github.com/shadowmoose/RedditDownloader/archive/995fc6876f111feab39040bf9aba0b47a87f4b06.zip).

Keep in mind, this release will convert your settings file & Manifest into new formats which are not backwards-compatible, so I suggest keeping a backup of those files somewhere in case you want to revert to an older RMD version in the meantime.

**Comment by JustCampMan** [comment]
Is that auth functionality related to 2FA? I was reading about that in another issue.

I tried to run the file like you said, but it isn't downloading anything.

![screenshot 115](https://user-images.githubusercontent.com/39931683/41003676-dbd5adc2-6930-11e8-9fce-59d6d6d4effd.png)
![screenshot 114](https://user-images.githubusercontent.com/39931683/41003677-dc638a3e-6930-11e8-9362-13050375f529.png)

The first image is how it stays, and the second image is the error it gives for a split second before the first image is displayed.

**Comment by shadowmoose** [comment]
Yeah, it's the 2FA fix slowing things down.

The error there is caused by the subreddit data field being missing in the settings file. I forgot about that one, and I'll have to roll out a fix that hardens Source configuration loading.

The Wizard is being phased out in favor of the WebUI, which is much better for editing settings. Thinking about it, if the WebUI works for you, you might actually be able to make the change to the Source there. It's disabled by default currently, so you'd have to edit the settings file and change ```interface > start_server``` to ```True```, then relaunch RMD without the ```-noui``` flag.

Either way, I'll commit a fix for your current issue soon.

**Comment by JustCampMan** [comment]
When you commit the fix, will it be available from the same link you previously posted or another one?

Also, is the settings file that I'm supposed to edit "settings.json"?

Thank you for your extremely quick replies.

**Comment by JustCampMan** [comment]
Well the WebUI worked, and all I had to do was specify the subreddit, and it started. Seeing that I have 392 subreddits with saved posts in each (don't judge :p), is there a way to automate it? So that it cycles through all my subs one by one.


![screenshot 118](https://user-images.githubusercontent.com/39931683/41007146-e41abede-693d-11e8-8d2e-146f91401b3e.png)

**Comment by shadowmoose** [comment]
Glad the WebUI worked.

As for automation, that's a harder one. How are you currently storing the subs? In a list, or just as you subscriptions?

From a quick search, I don't think Reddit has an API solution exposed to list all Subs you've upvoted or saved something in, so there's not an easy way to automatically detect which subs to scan. I could iterate the first 1000 posts and then scan those subs, but that's also not reliable enough.

I could make ```scan_sub``` accept a comma-separated list of subs to include. That might be easiest. Though I'm still going to search for a more elegant API solution.

**Comment by JustCampMan** [comment]
I'm storing all of my subreddits in which I saved posts as a .txt file. All you gotta do is click on the drop-down and select all, and paste into a txt file.

I was also trying to look for that API earlier, but couldn't find one. If you can incorporate the txt file solution for now, that'd be awesome.

Also (I know I keep asking too much again, and I apologize), is there a way to to unsave all the posts as RDM downloads them?

**Comment by shadowmoose** [comment]
For the file solution: I think I'll just add support for a comma-separated list, and you can paste it in there. There won't be a limit to the amount of subs you can filter by. It's pretty easy to reformat a text file into a list, and I'd like to keep all the configuration in one Menu/file.

As to removing posts: This feature is outside the scope of RMD. RMD will always be read-only when it comes to Reddit, because doing more would require extra permissions when authorizing. Also, it would open the door to nasty accidents when configured improperly. If there's enough of a use case for such a thing, I could easily roll up an extra utility in the ```tools``` directory.

**Comment by JustCampMan** [comment]
Yes exactly. Will it be like aww,funny,pics, etc etc or some other format? Please let me know when you merge this commit.

And yes you are right. If all the posts are unsaved automatically by mistake, it would be catastrophic.

**Comment by shadowmoose** [comment]
As of the new commit, 7022827b6a0ca8d4e944c308a792ef88c2541233, it should accept subreddits separated by comma values. You can download the frozen zip [here](https://github.com/shadowmoose/RedditDownloader/archive/7022827b6a0ca8d4e944c308a792ef88c2541233.zip).

**Comment by JustCampMan** [comment]
Should the text file be in the same folder as main.py?

Should the name be "scan_sub.txt"?

Should the text file have no spaces and just commas separating the names? Like (aww,pics,funny) or should there be spaces like (aww, pics, funny)?

**Comment by JustCampMan** [comment]
Ok so you just add the subreddits with only comas in between into settings.json. Worked like a charm.

Thank you!
